Versioning.md

```diff
@@ -0,0 +1,38 @@
+# iTwin.js versioning policies
+
+iTwin.js employs versioning policies to facilitate the following objectives:
+
+- Provide a broad set of stable, well-supported APIs.
+- Continuously deliver new features, bug fixes, and security updates.
+- Evolve APIs over time in response to user feedback.
+- Encourage users to keep up to date with the most recent versions.
+
+## API promotion process
+
+Each API exported by a package is marked with a [release tag](./docs/learning/guidelines/release-tags-guidelines.md) indicating its level of stability. A package's "public" API comprises all of its exported APIs that have been marked with the `@public` release tag. All other APIs are considered "non-public". The [package versioning](#package-versioning-policy) and [API deprecation](#api-deprecation-policy) policies apply only the a package's public API.
+
+`@internal` APIs are never intended to be used outside of the iTwin.js library itself. All other APIs are intended to ultimately become `@public` through the following process:
+
+- A highly experimental API may start out tagged as `@alpha`, indicating it is not yet mature enough for users to try out. This stage is usually skipped.
+- A new API, or one that has been promoted from `@alpha`, is tagged as `@beta` to invite users to experiment with it and provide feedback. The API may change in response to feedback from one release to the next.
+  - Note: extension APIs are tagged as `@preview` instead of `@beta`.
+- A `@beta` API that has been sufficiently tested is promoted to `@public` to indicate that it is now subject to [stability guarantees](#package-versioning-policy).
+- A `@public` API may be tagged as `@deprecated` indicating that it may later be removed according to the [deprecation policy](#api-deprecation-policy).
+
+Occasionally, a `@beta` or `@alpha` API may be abandoned rather than being promoted. Not all experiments are successful.
+
+In practice, `@beta` and `@public` APIs are typically designed in a way that promotes flexibility in the future - e.g., by accepting their arguments as "options" object types. This reduces the risk of breaking changes and deprecation.
+
+## Package versioning policy
+
+The individual packages that constitute the iTwin.js libraries use [semantic versioning](https://semver.org/) to convey information about how their public APIs change over time.
+
+- Patch releases occur frequently to fix bugs or security issues and have no impact on the public API.
```

Comment:
What do you think of having the description below rather than trying to convey the frequency of a particular release type?  I had started out with something similar to what you have but felt that the clarification on each was unneeded as the time is predictable and therefore not too useful to call out.


    MAJOR: Potentially contains backwards incompatible changes. A public API will not change in a backwards incompatible way unless it has been marked as deprecated for at minimum 1 major version. See [release tags](https://github.com/iTwin/itwinjs-backlog/issues/391#release-tags) for details on how to determine if an API is public.
    MINOR: Backwards compatible features, bug fixes and other enhancements to public APIs. However, in some cases beta and alpha APIs are subject to change. See [release tags](https://github.com/iTwin/itwinjs-backlog/issues/391#release-tags) for details on compatibility and how to determine if an API is public.
    PATCH: Backwards compatible bug fixes and security updates.


However we could remove the talk about release tags and their support in this case to merge it with yours. Something like,

    MAJOR: Potentially contains backwards incompatible changes. A public API will not change in a backwards incompatible way unless it has been marked as deprecated according to our `deprecation policy`.
    MINOR: Backwards compatible features, bug fixes and other enhancements to public APIs. May introduce new APIs, but never break any existing `@public` APIs.
    PATCH: Backwards compatible bug fixes and security updates. No impact on the `@public` API.

Reply:
Sure. I mainly want to convey that major releases are infrequent occurrences to assuage fears that we'll constantly be breaking APIs.

I'm moving this file to docs/learning and reworking it based on feedback.